draft-rescorla-tls-subcerts.md

```diff
@@ -280,18 +300,40 @@ Con:
 ## Option 2. Define a New Structure
 
 While X.509 forbids end-entity certificates from being used as issuers for other
-certificates, it is perfectly fine to use them to issue other signed objects.
-We could define a new signed object format that would encode only the semantics
-that are needed for this application.  For example, the TLS `digitally-signed`
-structure could be used:
+certificates, it is perfectly fine to use them to issue other signed objects as
+long as the certificate contains the digitalSignature key usage (RFC5280 section
+4.2.1.3). We could define a new signed object format that would encode only the
+semantics that are needed for this application.  For example, the TLS
+`digitally-signed` structure could be used:
 
 ~~~~~~~~~~
 digitally-signed struct {
   uint64 notBefore;
   uint64 notAfter;
   SignatureScheme algorithm;
+  ServerNameList serverName;
   opaque publicKey<0..2^24-1>;
-} SubCertificate;
+} DelegatedCredential;
+~~~~~~~~~~
+
+The ServerNameList can be defined as follows (similar to RFC 6066 section 3):
+~~~~~~~~~~
+struct {
+  NameType name_type;
+  select (name_type) {
+    case host_name: HostName;
+  } name;
+} ServerName;
+
+enum {
+  host_name(0), (255)
+} NameType;
+
+opaque HostName<1..2^16-1>;
+
+struct {
+  ServerName server_name_list<1..2^16-1>
```

Comment:
Nit: you need to have a length value in the front of ServerName. Otherwise you can't parse unexpected name types.